okhttp/src/commonJvmAndroid/kotlin/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -128,27 +128,40 @@ class CallServerInterceptor(
       exchange.responseHeadersEnd(response)
 
       response =
-        if (forWebSocket && code == 101) {
+        if (forWebSocket && code == HTTP_SWITCHING_PROTOCOLS) {
           // Connection is upgrading, but we need to ensure interceptors see a non-null response body.
           response.stripBody()
         } else {
-          val responseBody = exchange.openResponseBody(response)
-          response
-            .newBuilder()
-            .body(responseBody)
-            .trailers(
-              object : TrailersSource {
-                override fun peek() = exchange.peekTrailers()
+          if (code == HTTP_SWITCHING_PROTOCOLS &&
```

Comment:
We probably also need _something_ here to confirm it’s an HTTP/1 connection and not an HTTP/2 connection. We’ll implement that in a follow-up PR!

Reply:
According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Upgrade the `Upgrade` header is forbidden for HTTP/2. I didn't verify with the RFC, yet.
How strictly does OkHttp enforce usage conform to HTTP standards?

Also, the Websocket upgrade can be regarded as special case of the common HTTP/1.1 connection upgrade - is there a chance to upgrade HTTP/1.0 to Websocket?

Reply:
HTTP/2 verification is adressed in https://github.com/square/okhttp/pull/8969